__tests__/app.test.js

```diff
@@ -0,0 +1,304 @@
+            })
+    })
+    test("400 - responds with error message when only body is present without username", () => {
+        const badRequestComment = { body: "bad request comment" }
+        return request(app)
+            .post(`/api/articles/${articleId}/comments`)
+            .send(badRequestComment)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("bad request")
+        })
+    })
+    test("400 - responds with error message when article id is invalid", () => {
+        return request(app)
+            .post(`/api/articles/one/comments`)
+            .send(testComment)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("bad request")
+        })
+    })
+    test("404 - responds with error message when username does not exist", () => {
+        const noUserExistComment = { username: "unknownUser", body: "test" }
+        return request(app)
+            .post(`/api/articles/${articleId}/comments`)
+            .send(noUserExistComment)
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("no user exists")
+        })
+    })
+    test("404 - responds with error message when article id does not exist", () => {
+        return request(app)
+            .post(`/api/articles/99999/comments`)
+            .send(testComment)
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("no comments in article found with ID 99999")
+        })
+    })
+})
+describe('PATCH /api/articles/:article_id', () => {
+    const testPatch = { inc_votes: 1 }
+    const testPatchMinus = { inc_votes: -100 }
+    const errorPatch = { inc_votes: "one" }
+    const articleId1 = {article_id: 1,
+                        title: "Living in the shadow of a great man",
+                        topic: "mitch",
+                        author: "butter_bridge",
+                        body: "I find this existence challenging",
+                        created_at: 1594329060000,
+                        votes: 100,
+                        article_img_url: "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+                    }
+    test("200 - responds with updated article and new plus vote count", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .send(testPatch)
+            .expect(200)
+            .then(({ body }) => {
+                const updatedVotes = body.article.votes
+                const expectedVotes = articleId1.votes + testPatch.inc_votes
+                expect(updatedVotes).toBe(expectedVotes)
+        })
+    })
+    test("200 - responds with updated article and new minus vote count", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .send(testPatchMinus)
+            .expect(200)
```

Comment:
This is really similar to the test above, you could make it DRYer by making it one test with two expect assertions